module/data/item/templates/activities.mjs

```diff
@@ -90,7 +90,7 @@ export default class ActivitiesTemplate extends SystemDataModel {
    * @type {boolean}
    */
   get hasLimitedUses() {
-    return this.uses.max > 0;
+    return !!this.uses.max;
```

Comment:
You'll need to check source data for this, because during data preparation we resolve the formula and if no roll data is available it defaults to `0`.